diff --git a/Library/Contributions/brew_bash_completion.sh b/Library/Contributions/brew_bash_completion.sh
index f3c685d..fce18d4 100644
--- a/Library/Contributions/brew_bash_completion.sh
+++ b/Library/Contributions/brew_bash_completion.sh
@@ -93,6 +93,11 @@ __brew_complete_tapped ()
     __brewcomp "$(\ls $(brew --repository)/Library/Taps 2>/dev/null | sed 's/-/\//g')"
 }
 
+__brew_complete_anytapped ()
+{
+    __brewcomp "$(\ls $(brew --repository)/Library/Taps 2>/dev/null)"
+}
+
 _brew_complete_tap ()
 {
     local cur="${COMP_WORDS[COMP_CWORD]}"
@@ -400,6 +405,8 @@ _brew ()
         __brewcomp "
             --cache --cellar --config
             --env --prefix --repository
+            any-tap
+            any-untap
             audit
             cat
             cleanup
@@ -462,6 +469,7 @@ _brew ()
     tap)                        _brew_complete_tap ;;
     uninstall|remove|rm)        _brew_uninstall ;;
     unpin)                      __brew_complete_formulae ;;
+    any-untap)                  __brew_complete_anytapped ;;
     untap)                      __brew_complete_tapped ;;
     update)                     _brew_update ;;
     uses)                       _brew_uses ;;
