diff --git Library/Contributions/brew_bash_completion.sh Library/Contributions/brew_bash_completion.sh
index 0695674..4e64bee 100644
--- Library/Contributions/brew_bash_completion.sh
+++ Library/Contributions/brew_bash_completion.sh
@@ -92,6 +92,11 @@ __brew_complete_tapped ()
     __brewcomp "$(\ls $(brew --repository)/Library/Taps 2>/dev/null | sed 's/-/\//g')"
 }
 
+__brew_complete_anytapped ()
+{
+    __brewcomp "$(\ls $(brew --repository)/Library/Taps 2>/dev/null)"
+}
+
 __brew_complete_taps ()
 {
     if [[ -z "$__brew_cached_taps" ]]; then
@@ -380,6 +385,8 @@ _brew ()
         __brewcomp "
             --cache --cellar --config
             --env --prefix --repository
+            any-tap
+            any-untap
             audit
             cat
             cleanup
@@ -438,6 +445,7 @@ _brew ()
     search|-S)                  _brew_search ;;
     tap)                        __brew_complete_taps ;;
     uninstall|remove|rm)        _brew_uninstall ;;
+    any-untap)                  __brew_complete_anytapped ;;
     untap)                      __brew_complete_tapped ;;
     update)                     _brew_update ;;
     uses)                       _brew_uses ;;
